Code/Source/Lidar/PipelineGraph.cpp

```diff
@@ -177,17 +174,28 @@ namespace RGL
 
     bool PipelineGraph::GetResults(RaycastResults& results)
     {
-        int32_t resultSize = -1;
-        RGL_CHECK(rgl_graph_get_result_size(m_nodes.m_pointsFormat, rgl_field_t::RGL_FIELD_DYNAMIC_FORMAT, &resultSize, nullptr));
-
-        if (resultSize <= 0)
+        bool success = true;
+        for (rgl_field_t field : results.m_fields)
         {
-            return false;
+            switch (field)
+            {
+            case RGL_FIELD_IS_HIT_I32:
+                success = success && GetResult(results.m_isHit, RGL_FIELD_IS_HIT_I32);
```

Comment:
Are you aware that this will prevent calling every GetResult after the first fail? Because, when

state = false && Function();

Function is not called at all. This is not necessarily a bug or a mistake - depends on your intentions.

Reply:
Good catch, I missed the fact that this is lazy logic.

Reply:
If one ore more of these return false, there's no reason to collect the results since the task has already failed.